xonsh/xoreutils/uname.py

```diff
@@ -0,0 +1,209 @@
+import platform
+import sys
+from xonsh.xoreutils.util import arg_handler
+
+
+def uname(args, stdin, stdout, stderr):
+    """
+    Print certain system information.
+
+    :param args: Arguments like -a
+    :type args: list or None
+    :param stdin: The standard input
+    :type stdin: sys.stdin
+    :param stdout: The standard output
+    :type stdout: sys.stdout
+    :param stderr: The standard output
+    :type stderr: sys.stderr
+    """
+    if stdout is None:
+        stdout = sys.stdout
+
+    opts = _uname_parse_args(args)
+
+    if opts is None:
+        opts = {
+            "kernel_name": False,
+            "nodename": False,
+            "kernel_release": False,
+            "kernel_version": False,
+            "machine": False,
+            "processor": False,
+            "hardware_platform": False,
+            "operating_system": False,
+            "version": False,
+            "help": False,
+        }
+
+    # control
+    everything_is_false = True
+    for key, value in list(opts.items()):
+        if key not in [
```

Comment:
This should be a set, not a list. It should probably also be defined at the module level